File: src/lib/grades.ts
import {
  Cpu, Code2, Globe2, Layers, Sparkles, TerminalSquare,
  BarChart3, GraduationCap, Compass, BookOpen, Scale, PenTool, ShieldCheck,
  type LucideIcon,
} from "lucide-react";

export type GradeId =
  | "core" | "engineer" | "researcher" | "architect" | "creator" | "operator"
  | "analyst" | "educator" | "strategist" | "scholar" | "legal" | "designer" | "guardian";

export const GRADE_LIST: Array<{ id: GradeId; label: string; blurb: string; detail: string; icon: LucideIcon }> = [
  { id: "core", label: "Core", blurb: "Balanced general intelligence", detail: "Your default assistant for everyday questions, writing, and quick help.", icon: Cpu },
  { id: "engineer", label: "Engineer", blurb: "Deep coding, debugging, refactors", detail: "Full programs, bug fixes, and code reviews across any language.", icon: Code2 },
  { id: "researcher", label: "Researcher", blurb: "Live web evidence, cited", detail: "Pulls current information from the web and cites sources.", icon: Globe2 },
  { id: "architect", label: "Architect", blurb: "Systems, infra, scale", detail: "System design, architecture decisions, and scaling strategy.", icon: Layers },
  { id: "creator", label: "Creator", blurb: "Writing, naming, narrative", detail: "Copywriting, brand names, stories, and creative content.", icon: Sparkles },
  { id: "operator", label: "Operator", blurb: "Shell, ops, automation", detail: "Exact terminal commands, scripts, and automation workflows.", icon: TerminalSquare },
  { id: "analyst", label: "Analyst", blurb: "Data, metrics, forecasting", detail: "Breaks down numbers, spreadsheets, and business metrics.", icon: BarChart3 },
  { id: "educator", label: "Educator", blurb: "Clear, step-by-step teaching", detail: "Explains concepts simply, like a patient tutor.", icon: GraduationCap },
  { id: "strategist", label: "Strategist", blurb: "Business planning & decisions", detail: "Helps think through strategy, tradeoffs, and next moves.", icon: Compass },
  { id: "scholar", label: "Scholar", blurb: "Deep, rigorous long-form analysis", detail: "Thorough, well-reasoned answers for complex topics.", icon: BookOpen },
  { id: "legal", label: "Legal", blurb: "Contracts & compliance awareness", detail: "Reviews and drafts documents with legal structure in mind (not a lawyer).", icon: Scale },
  { id: "designer", label: "Designer", blurb: "UI/UX & visual design feedback", detail: "Product design critique, layout ideas, and visual polish.", icon: PenTool },
  { id: "guardian", label: "Guardian", blurb: "Security review & best practices", detail: "Reviews code and systems for vulnerabilities and hardening.", icon: ShieldCheck },
];

File: flutter/lib/core/grades.dart
class Grade {
  final String id;
  final String label;
  final String blurb;
  final String system;
  final int maxTokens;

  const Grade({
    required this.id,
    required this.label,
    required this.blurb,
    required this.system,
    required this.maxTokens,
  });
}

const grades = <Grade>[
  Grade(id: 'core', label: 'Core', blurb: 'Balanced general intelligence',
    system: 'You are JagX AI Core v1.1.2. Be direct, precise, and useful. Think before answering.', maxTokens: 2200),
  Grade(id: 'engineer', label: 'Engineer', blurb: 'Deep coding, debugging, refactors',
    system: 'You are JagX AI Engineer. Ship complete, runnable code. Explain only what is needed.', maxTokens: 3200),
  Grade(id: 'researcher', label: 'Researcher', blurb: 'Evidence-led synthesis with sources',
    system: 'You are JagX AI Researcher. Use live web context when provided. Cite sources as [n].', maxTokens: 2600),
  Grade(id: 'architect', label: 'Architect', blurb: 'Systems, infra, scale',
    system: 'You are JagX AI Architect. Reason about tradeoffs, failure modes, and scale.', maxTokens: 2400),
  Grade(id: 'creator', label: 'Creator', blurb: 'Writing, naming, narrative',
    system: 'You are JagX AI Creator. Write with taste. Be original.', maxTokens: 2000),
  Grade(id: 'operator', label: 'Operator', blurb: 'Terminal, shell, automation',
    system: 'You are JagX AI Operator. Give exact commands and scripts.', maxTokens: 1800),
  Grade(id: 'analyst', label: 'Analyst', blurb: 'Data, metrics, forecasting',
    system: 'You are JagX AI Analyst. Use tables and a clear takeaway.', maxTokens: 2200),
  Grade(id: 'educator', label: 'Educator', blurb: 'Step-by-step teaching',
    system: 'You are JagX AI Educator. Teach patiently from first principles.', maxTokens: 2000),
  Grade(id: 'strategist', label: 'Strategist', blurb: 'Plans and decisions',
    system: 'You are JagX AI Strategist. Frame goals, constraints, next actions.', maxTokens: 2000),
  Grade(id: 'scholar', label: 'Scholar', blurb: 'Long-form rigorous analysis',
    system: 'You are JagX AI Scholar. Be thorough, structured, and precise.', maxTokens: 3200),
  Grade(id: 'legal', label: 'Legal', blurb: 'Contracts and compliance awareness',
    system: 'You are JagX AI Legal. Always state you are not a lawyer and this is not legal advice.', maxTokens: 2200),
  Grade(id: 'designer', label: 'Designer', blurb: 'UI/UX critique',
    system: 'You are JagX AI Designer. Be specific about hierarchy and usability.', maxTokens: 2000),
  Grade(id: 'guardian', label: 'Guardian', blurb: 'Security review',
    system: 'You are JagX AI Guardian. Rank issues by severity; propose fixes.', maxTokens: 2400),
];

Grade gradeById(String id) =>
    grades.firstWhere((g) => g.id == id, orElse: () => grades.first);
